Implement Podinfo with per-environment Values

glarizza opened the pull request:
PROBLEM:

We've added code to the repo demonstrating a way to handle environments,
but we haven't yet tackled the problem of a component with imported
default values that has a different value on a per-environment basis.

SOLUTION:

Add a `podinfo` component that uses the correct environment namespace,
then update the configuration to ensure there are three replicas in
`prod` and a single replica in other environments.

OUTCOME:

A very basic example showing different Helm chart values on
a per-environment basis.

glarizza committed:
Implement Podinfo with per-environment Values

PROBLEM:

We've added code to the repo demonstrating a way to handle environments,
but we haven't yet tackled the problem of a component with imported
default values that has a different value on a per-environment basis.

SOLUTION:

Add a `podinfo` component that uses the correct environment namespace,
then update the configuration to ensure there are three replicas in
`prod` and a single replica in other environments.

OUTCOME:

A very basic example showing different Helm chart values on
a per-environment basis.

jeffmccune commented on projects/bank-frontend/components/podinfo/podinfo.cue:
```diff
@@ -0,0 +1,27 @@
+    Values: replicaCount: [
+        if EnvironmentName =~ "^prod" {3},
+        1,
+    ][0] // Select the first element in the list
```
Let's not use a list comprehension like this, while it works and I eventually figured it out, it's not what I had in mind.

To vary by prod or nonprod, I was thinking we look things up in a struct, like this:

```cue
#TierValues: [TIER=string]: replicaCount: int & >=0

PodInfo: TierValues: #TierValues
PodInfo: TierValues: nonprod: replicaCount: 1
PodInfo: TierValues: prod: replicaCount: 3
```

Then in `projects/bank-frontend/components/podinfo/podinfo.cue` we can dig into that struct:

```cue
let TIER = BankOfHolos.Environments[EnvironmentName].tier
Podinfo: Values: Podinfo.TierValues[TIER]
```

jeffmccune reviewed (changes requested):
One comment on how we look up the data.

I'm also working on re-structuring the bank projects, so I'll help update and resolve the conflicts in `./bank-projects.cue`

jeffmccune commented:
@glarizza Let me know when this is ready again.  FYI, as of 157f49e I don't see where podinfo gets added to the Platform though.

```
❯ git diff main --stat '**/*.cue' '*.cue'
 organization-gary.cue                                            |   9 +
 projects/bank-frontend/components/podinfo/podinfo.cue            |  27 +
 projects/bank-frontend/components/podinfo/podinfo.values.gen.cue | 211 +
 3 files changed, 247 insertions(+)
```

Ideally we could opt-in with a build tag like we did for ChartMuseum which I mentioned in the [Private Helm](https://holos.run/docs/v1alpha5/topics/private-helm/) topic.

```cue
@if(ChartMuseum)
package holos

Projects: holos: #ProjectBuilder & {
        team: "holos-authors"

        namespaces: holos:           _
        _components: "chart-museum": _
}
// Register the HTTPRoute to the backend Service
HTTPRoutes: charts: _backendRefs: chartmuseum: namespace: "holos"
```

glarizza commented:
Okay, ready for another review:

* Updated `bank-projects.cue` to ensure `podinfo` gets added to all clusters and environments
* Updated `podinfo.cue` to manually specify the Helm chart name (which doesn't need the environment name)
* Updated `podinfo.cue` to specify the tier-specific values using your recommendation above
